--- /opt/victronenergy/gui/qml/PageAcInSetup_.qml	
+++ /opt/victronenergy/gui/qml/PageAcInSetup.qml
@@ -84,6 +84,39 @@ MbPage {
 		return qsTr("Unknown");
 	}
 
+	VBusItem {
+		id: meterCount
+		bind: Utils.path(root.bindPrefix, "/MeterCount")
+		onValueChanged: phase.possibleValues = getPhaseList(value)
+	}
+
+	function getPhaseList(count)
+	{
+		var options = [];
+		for (var i = 1; i < 4; i++) {
+			var params = {
+				"description": "L" + i,
+				"value": i,
+			}
+			options.push(mbOptionFactory.createObject(root, params));
+		}
+		
+		for (var i = 4; (i < 7) && count === 3; i++) {
+			var params = {
+				"description": "L" + ((i-1) % 3 + 1) + "/L" + ((i) % 3 + 1) +"/L" + ((i+1) % 3 + 1),
+				"value": i,
+			}
+			options.push(mbOptionFactory.createObject(root, params));
+		}
+		
+		return options;
+	}
+
+	VBusItem {
+		id: instance
+		bind: Utils.path(root.bindPrefix, "/DeviceInstance")
+	}
+
 	VBusItem {
 		id: allowedRoles
 		bind: Utils.path(root.bindPrefix, "/AllowedRoles")
@@ -140,6 +173,68 @@ MbPage {
 			show: productId == em24ProductId && em24Locked()
 		}
 
+		/* Shelly settings */
+
+		MbItemOptions {
+			id: phase
+			show: productId == 0xFFE0
+			description: qsTr("Phase")
+			bind: Utils.path("com.victronenergy.settings/Settings/Shelly/", instance.value, "/Phase")
+			readonly: false
+			editable: true
+		}
+
+		MbSpinBox {
+			id: meterIndex
+			show: productId == 0xFFE0 && meterCount.value > 1 && phase.value < 4
+			description: qsTr("Meter Index")
+			item {
+				bind: Utils.path(root.bindPrefix, "/MeterIndex")
+				decimals: 0
+				step: 1
+				max: meterCount.value - 1
+				min: 0
+			}
+		}
+
+		MbEditBoxIp {
+			show: productId == 0xFFE0
+			description: qsTr("IP Address")
+			item: VBusItem {
+				id: shellyIpaddress
+				isSetting: true
+				bind: Utils.path("com.victronenergy.settings/Settings/Shelly/", instance.value, "/Url")
+			}  
+		}
+		
+		MbEditBox {
+			show: productId == 0xFFE0
+			description: qsTr("User Name")
+			maximumLength: 35
+			item: VBusItem {
+				id: shellyUserName
+				isSetting: true
+				bind: Utils.path("com.victronenergy.settings/Settings/Shelly/", instance.value, "/Username")
+			} 
+		}
+		
+		MbEditBox {
+			show: productId == 0xFFE0
+			description: qsTr("Password")
+			maximumLength: 35
+			item: VBusItem {
+				id: shellyPassword
+				isSetting: true
+				bind: Utils.path("com.victronenergy.settings/Settings/Shelly/", instance.value, "/Password")
+			} 
+		}
+
+		MbSwitch {
+			id: temperatureSensor
+			bind: Utils.path("com.victronenergy.settings/Settings/Shelly/", instance.value, "/TemperatureSensor") 
+			name: qsTr("Show Temperature")
+		}
+
 		/* Smappee settings */
 
 		MbItemOptions {
